### DISCUSSION TITLE
Current Status of SqlKata

### DISCUSSION THREAD

**Discussion by jonreis** [sqlkata/querybuilder#728] [Q&A]
Hello Team, I am looking at using QueryBuilder in a new project but I don't see much activity going on with the library. I can see there is a version 3.0 that has been in beta since 2022.

Overall the project looks pretty useful. Would you recommend using it for new projects, or is it being abandoned?

Thank you in advance.

**Comment by gaperton** [comment]
Anyone interested in creating a fork and starting to fix bugs and make improvements? Perhaps, there are some forks already?

I need it for my project and would like to join the maintainer community.

**Reply by gaperton** [reply]
@ahmad-moussawi Do you still maintain this thing? Maybe you want some help? I hate to fork good projects.

**Reply by JordanMarr** [reply]
Ideally, @ahmad-moussawi could add a few volunteers to help with PRs and maintenance. 
But without any response, creating a fork seems acceptable. You did your due diligence by asking first. 👍
I'm sure many people would greatly appreciate having a place to submit PRs.

**Reply by deadwards90** [reply]
I'm interested in creating a fork. I can deal with the SQL Server side, might need some assistance on the other SQL dialects but I'm happy to run things.

Priorities would be

1. Getting the current beta version released.
2. Fix any pressing bugs and just stabilising the release
3. Adding new features where we can.

I don't want to see this project die and would happily just come on as a maintainer if that was easier but that would be up to @ahmad-moussawi. Don't want to step on any toes.

**Reply by ahmad-moussawi** [reply]
Hey everyone see my answer below, you can always fork the project if you see that would help, 
but in terms of project benefits, concentrating the effort on one project would be more beneficial,
if you have the time to help, there are a lot of issues/bug fixes that need to be fixed.

I will prioritize the issue list so this facilitates picking an issue.

**Reply by deadwards90** [reply]
I'd much prefer not to fork it, I don't want to split the community 😄 

I'd be more than happy to start picking up issues though.

**Comment by ahmad-moussawi** [comment]
Hey, I will post a more detailed announcement soon.

TLDR.  We live in a dramatic situation in Lebanon, where I live. however the project is still active, and major issues and bugs have always been addressed.

That being said, you are always welcome to create your fork if you feel you have a specific request that can't be merged.

**Reply by JordanMarr** [reply]
So sorry you are dealing with that problem. 
Stay safe. 🙏

**Reply by chris-tophe** [reply]
So did you consider to add volunteer to maintain the project ?

**Reply by ceyhunn** [reply]
This project is so helpful for building query, especially dynamic queries. So it would be nice to go on to maintain the project.

**Reply by faddiv** [reply]
What is the situation with the project? I didn't see anything happening for a long time.

**Reply by ahmad-moussawi** [reply]
Thanks, everyone, 
things are getting better over time, 
as I said before, the project will still be supported, because: 
 - I use it on my own projects
 - There is no similar alternative in the dotnet world.
 So I have no plan to stop supporting it.
 
 That being said, volunteering/sponsoring this project is always welcomed.
 
a new version [4.0](https://github.com/sqlkata/querybuilder/releases/tag/v4.0.0) was released a few days ago, dropping support for the legacy `.netfx` to reduce the friction and help focus on new features and improving the quality of this library.

**Reply by michelbieleveld** [reply]
Also asking if you have considered adding a volunteer to maintain the project, where maintaining means allowed to be able to merge requests. I do respect the situation you are in, but adding a volunteer is exactly so you can have more support.
